src/native/minipal/gethomedir.h

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef HAVE_MINIPAL_GETHOMEDIR_H
+#define HAVE_MINIPAL_GETHOMEDIR_H
+
+#include <errno.h>
+#include <pwd.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Returns the full path to the home directory.
+// The caller is responsible for releasing the buffer. Returns null on error.
+static inline char* minipal_gethomedir(void)
+{
+    char* homedir = getenv("HOME");
```

Comment:
What if HOME was changed via Environment.SetEnvironmentVariable?  While maybe not a common or recommended thing to do, this would represent a breaking change.

Reply:
I didn't realize this scenario and there is also a test for it. getenv will return the actual value and process change won't be reflected. Definitely a breaking change. :(